js/repeater.js

```diff
@@ -550,42 +606,35 @@
 					} else {
 						this.$nextBtn.focus();
 					}
-
+				} else if (this.$prevBtn.is(':disabled')) {
+					// if you can't focus, go the other way
+					this.$nextBtn.focus();
 				} else {
-					if (this.$prevBtn.is(':disabled')) {
-						// if you can't focus, go the other way
-						this.$nextBtn.focus();
-					} else {
-						this.$prevBtn.focus();
-					}
-
+					this.$prevBtn.focus();
 				}
-
 			}
 		},
 
-		previous: function () {
-			var d = 'disabled';
-			this.$nextBtn.attr(d, d);
-			this.$prevBtn.attr(d, d);
+		previous: function previous () {
+			this.$nextBtn.attr('disabled', 'disabled');
+			this.$prevBtn.attr('disabled', 'disabled');
 			this.pageIncrement = -1;
 			this.$element.trigger('previousClicked.fu.repeater');
 			this.render({
 				pageIncrement: this.pageIncrement
 			});
 		},
 
-		render: function (options) {
-			var self = this;
+		// This functions more as a "pre-render" than a true "render"
+		render: function render (opts) {
+			this.disable();
```

Comment:
I usually think of statements after declarations, but I guess there could be enabled UI elements for that 10ms or so.